src/main/java/com/github/bpiatek/bbghbackend/model/article/search/ArticleSearcherImpl.java

```diff
@@ -0,0 +1,169 @@
+package com.github.bpiatek.bbghbackend.model.article.search;
+
+import com.github.bpiatek.bbghbackend.model.comment.CommentOpinionStatus;
+import com.github.bpiatek.bbghbackend.util.sql.SqlSelectStatement;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Service;
+
+import javax.persistence.*;
+import java.math.BigInteger;
+import java.sql.Timestamp;
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Błażej Rybarkiewicz <b.rybarkiewicz@gmail.com>
+ */
+@Service
+public class ArticleSearcherImpl implements ArticleSearcher {
+
+  @PersistenceContext
+  private final EntityManager em;
+
+  public ArticleSearcherImpl(EntityManager entityManager) {
+    this.em = entityManager;
+  }
+
+  @Override
+  public List<ArticleSearchFilter> getFiltersFromSearchQuery(String query) {
+    List<ArticleSearchFilter> filterList = new ArrayList<>();
+    if (query == null) {
+      return filterList;
+    }
+    query = query.replaceAll("\\s+","").toLowerCase();
+    Pattern pattern = Pattern.compile("(\\w+?)(:|<|>|=)([^,]+?),");
+    Matcher matcher = pattern.matcher(query + ",");
+    while (matcher.find()) {
+      filterList.add(new ArticleSearchFilter(matcher.group(1), matcher.group(2), matcher.group(3)));
+    }
+    return filterList;
+  }
+
+  @Override
+  public Page<ArticleSearchResult> find(Pageable pageable, List<ArticleSearchFilter> filterList) {
+    // Using LinkedHashMap because we need to keep elements order
+    Map<String,String> fields = new LinkedHashMap<>() {{
+      String selectCountByStatus = "(SELECT COUNT(id) FROM comment WHERE article_id = article.id AND comment_opinion_status = '%s')";
+      // fields are lowercase to simplify filter step
+      // alias, field
+      put("id", "id");
+      put("url", "url");
+      put("title", "title");
+      put("creationdate", "creation_date");
+      put("content", "content");
+      put("comments", "(SELECT COUNT(id) FROM comment WHERE article_id = article.id)");
+      put("positive", String.format(selectCountByStatus, CommentOpinionStatus.POSITIVE));
```

Comment:
w ogole zamiast tylu query to mozna chyba zalatwic jednym takim: (wiadomo, ze statusow mozesz dac wiecej)
`SELECT 
	SUM(comment_opinion_status = 'NEGATIVE') AS 'negative_comments',
    SUM(comment_opinion_status = 'POSITIVE') AS 'opinions'
FROM testapp.comment 
WHERE article_id = 6;`

i wtedy do jednego obiektu zapisujesz ilosc danych wartosci w kolumnie

Reply:
Powiem szczerze, że nie wiem jak miałbym użyć tych sum potem i zrobi filtrowanie i order :smile: .